compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3670,6 +3670,10 @@ private boolean equalityIntersectionExists(Set<BType> lhsTypes, Set<BType> rhsTy
     }
 
     boolean validNumericTypeExists(BType type) {
+
+        if (type.isNullable() && type.tag != TypeTags.ANY && type.tag != TypeTags.NIL) {
+            type = ((BUnionType) type).getMemberTypes().iterator().next();
+        }
```

Comment:
Is this the correct place for this change? Shouldn't this method be called with the non-nil type(s) instead?

Reply:
Since the method checks whether the type is valid numeric type or not, I tried to lift the nil of option type and check whether the type is valid or not